src/plugins/event-plugins/ResourcePlugin.ts

```diff
@@ -90,8 +90,17 @@ export class ResourcePlugin extends InternalPlugin {
     }: PerformanceResourceTiming): void => {
         const pathRegex =
             /.*\/application\/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\/events/;
-        const entryUrl = new URL(name);
+        let entryUrl: URL;
+        try {
+            entryUrl = new URL(name);
+        } catch {
+            // Ignore failures where an invalid string causes the URL
+            // construction to fail. Otherwise RUM failures are logged as
+            // an error event.
+            return;
```

Comment:
issue: If `name` is an invalid URL, we still want to record the event.
```suggestion
```

Reply:
@qhanam I was going off the suggestions in the previous version of changes:
> suggestion: Since this is the only place we use entryUrl, we could (1) modify this condition to only run when entryUrl is truthy, and (2) catch and ignore the error thrown by new URL(name).

Can you please clarify by providing the try/catch block you intended?

Reply:
In the current revision, we catch the error, but instead of ignoring it we exit the function without having recorded an event. The suggested change yields the intended behavior.

Reply:
I think see what Quinn is saying. Raised a PR

https://github.com/shubhsheth/aws-rum-web/pull/2